content/en/docs/reference/access-authn-authz/authentication.md

```diff
@@ -1198,6 +1170,181 @@ even if it is allowed by RBAC configuration.
+
+`--authentication-token-webhook-config-file`
+: The path to a kubeconfig format client configuration, that specifies how the API server authenticates when making HTTP callouts.
+
+`--authentication-token-webhook-version`
+: The API version of TokenReview to use when making HTTP callouts to check tokens.
+
+#### Web request authentication configuration {#api-server-authn-config-cli-reverse-proxy}
+
+<!-- trailing whitespace (exactly two spaces) is significant in the following list -->
+
+{{< caution >}}
+You should read the documentation about configuring an [authenticating proxy](#authenticating-proxy)
+before you specify these command line arguments, as there is important information security advice
+that you must follow.
+{{< /caution >}}
+
+`--requestheader-client-ca-file`
+: _Required._
+  Path to a PEM-encoded certificate bundle.  
+  A valid [client certificate](#reverse-proxy-client-certificate)
+  must be presented and validated against the certificate authorities in the specified file before the
+  request headers are checked for user names.
+
+`--requestheader-allowed-names`
+: _Optional._ Comma-separated list of Common Name values (CNs).  
+  If set, a valid client
+  certificate with a CN in the specified list must be presented before the request headers are checked
+  for user names. If empty, any CN is allowed.
+
+`--requestheader-username-headers`
+: _Required; case-insensitive._ Header names to check, in order, for the user identity.  
+  The first header containing a value is used as the username.
+
+`--requestheader-group-headers`
+: _Optional; case-insensitive._
+  Header names to check, in order, for the user's groups.  
+  `X-Remote-Group` is suggested.
+  All values in all specified headers are used as group names.
+
+`--requestheader-extra-headers-prefix`
+: _Optional; case-insensitive._
+  Header prefixes to look for to determine extra information about the user.  
+  `X-Remote-Extra-` is suggested.
+  Extra data is typically used by the configured authorization plugin(s).
+  Any headers beginning with any of the specified prefixes have the prefix removed.
+  The remainder of the header name is lowercased and [percent-decoded](https://tools.ietf.org/html/rfc3986#section-2.1)
+  and becomes the extra key, and the header value is the extra value.
+
+
+### Configuration via configuration file {#api-server-authn-config-file}
+
+{{< feature-state feature_gate_name="StructuredAuthenticationConfiguration" >}}
+
+When you specify the `--authentication-config` command line argument to the kube-apiserver, the API server
+loads a file at the path you specify, and uses the contents of that file to configure authentication.
+
+#### Configuration file path {#api-server-authn-config-cli-general}
+
+`--authentication-config`
+: This special command line argument specifies that you want to [configure authentication using a configuration file](#api-server-authn-config-file).
```

Comment:
The contents of this file can be changed while the apiserver is running and is re-read. Modifications to this file should be done in an atomic way (writing to a peer temporary file, then renaming the temporary file to replace this file, etc).